test-docker.sh

```diff
@@ -1,47 +1,101 @@
+PHPMYADMIN_PORT=${TESTSUITE_PORT:=80}
+PHPMYADMIN_URL=http://$PHPMYADMIN_HOSTNAME:$PORT/
+
+# Color text output
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m' # No Color
 
-# Wait for container to start
 ret=0
+# Check if script is running inside container
+if [ -f /.dockerenv ] ; then
+    echo "Tests running inside container..."
+
+    # Set database environment
+    PHPMYADMIN_DB_HOSTNAME=${PMA_HOST:=localhost}
+    PHPMYADMIN_DB_PORT=${PMA_PORT:=3306}
+    PHPMYADMIN_DB_URL=http://$PHPMYADMIN_DB_HOSTNAME:$PHPMYADMIN_DB_PORT/
+
+    # Wait for database to start
+    TIMEOUT=0
+    while ! curl "$PHPMYADMIN_DB_URL" &>/dev/null; do
+        echo "Waiting for ${PHPMYADMIN_DB_HOSTNAME} database start..."
+        sleep 10
+        TIMEOUT=$((TIMEOUT + 1))
+        if [ $TIMEOUT -gt 3 ] ; then
+            echo "Failed to connect ${PHPMYADMIN_DB_HOSTNAME} database!"
+            echo -e "Result of ${PHPMYADMIN_DB_HOSTNAME} tests: ${RED}FAILED${NC}"
+            ret=1
+            exit 1
+        fi
+    done
+
+    # Install mechanize via pip package manager
+    if ! python -c "import mechanize" &> /dev/null ; then
+        pip install --user mechanize
+        ret=$?
+    fi
+
+    # Install html5lib via pip package manager
+    if ! python -c "import html5lib" &> /dev/null ; then
+        pip install --user html5lib
+        ret=$?
+    fi
+else
+    echo "Tests running outside container..."
+    docker ps -a
+    COMMAND_HOST="docker exec ${NAME}"
+fi
+
+# Wait for container to start
 TIMEOUT=0
-while ! docker exec "$NAME" ps aux | grep -q nginx ; do
-    echo 'Waiting for start...'
+while ! $COMMAND_HOST ps aux | grep -q nginx ; do
+    echo "Waiting for PHPMyAdmin start..."
     sleep 1
     TIMEOUT=$((TIMEOUT + 1))
     if [ $TIMEOUT -gt 10 ] ; then
-        echo "Failed to wait!"
+        echo "Failed to connect PHPMyAdmin!"
+        echo -e "Result of ${PHPMYADMIN_DB_HOSTNAME} tests: ${RED}FAILED${NC}"
         ret=1
         exit 1
     fi
 done
 
 # Perform tests
 if [ $ret -eq 0 ] ; then
-    python phpmyadmin_test.py --url "$URL" --username root --password $TESTSUITE_PASSWORD $SERVER
+    python phpmyadmin_test.py --url "$PHPMYADMIN_URL" --username root --password $TESTSUITE_PASSWORD $SERVER
     ret=$?
 fi
 
 # Show debug output in case of failure
 if [ $ret -ne 0 ] ; then
-    curl "$URL"
-    docker ps -a
-    docker exec "$NAME" ps faux
-    docker exec "$NAME" cat /var/log/php-fpm.log
-    docker exec "$NAME" cat /var/log/nginx-error.log
-    docker exec "$NAME" cat /var/log/supervisord.log
+    curl "$PHPMYADMIN_URL"
+    $COMMAND_HOST ps faux
+    $COMMAND_HOST cat /var/log/php-fpm.log
+    $COMMAND_HOST cat /var/log/nginx-error.log
+    $COMMAND_HOST cat /var/log/supervisord.log
+    echo -e "Result of ${PHPMYADMIN_DB_HOSTNAME} tests: ${RED}FAILED${NC}"
     exit $ret
 fi
 
-# List processes
-docker exec "$NAME" ps faux
+echo -e "Result of ${PHPMYADMIN_DB_HOSTNAME} tests: ${GREEN}SUCCESS${NC}"
```

Comment:
The `echo -e` does not work in many /bin/sh implementations, it's better to avoid it (see Travis log for example).